### PR TITLE
Listen bucket notification for multiple prefixes/suffixes

### LINKED ISSUE #2871: Feature request: Listen bucket notification for multiple suffixes
Currently, listenBucketNotification only accepts single 'suffix', single 'prefix' query params. We should take a list and allow listening on them.

### PR DESCRIPTION
**Pull request by wendigo**
Fixes #2871

### CODE REVIEW

**Comment by harshavardhana** [comment]
This is good Thanks. Will review shortly.

File: cmd/bucket-notification-handlers.go [hunk 1]
```diff
@@ -250,10 +250,20 @@ func (api objectAPIHandlers) ListenBucketNotificationHandler(w http.ResponseWrit
...
+	prefixes, suffixes, events := getListenBucketNotificationResources(r.URL.Query())
+
+	for _, prefix := range prefixes {
+		if !IsValidObjectPrefix(prefix) {
+			writeErrorResponse(w, r, ErrFilterValueInvalid, r.URL.Path)
+			return
+		}
+	}
+
+	for _, suffix := range suffixes {
```
Comment (harshavardhana) [inline cmd/bucket-notification-handlers.go]: We could move this into a single function. Which validates and returns 'true/false' like following. 

```
validateFilterValues(values []string) APIErrorCode {
      for _, value := range values {
               if !IsValidObjectPrefix(value) {
                         return ErrFilterValueInvalid
               } 
      }
      return ErrNone
}
```

Now you can use this cleanly 

```
if apiErr := validateFilterValues(prefixes); apiErr != ErrNone {
              writeErrorResponse(w, r, apiErr, r.URL.Path)
}
```

This way you can add test cases for "validateFilterValues()"

Comment (wendigo) [reply cmd/bucket-notification-handlers.go]: Sure, refactored + added tests for new function

**Comment by wendigo** [comment]
Thx! :)

**Pull request merged**